### ISSUE TITLE
Favorite Projects - Add Favorite Dropdown to Project Detail Page Actions

### ISSUE THREAD

**Issue by RChandler234** [Northeastern-Electric-Racing/FinishLine#990]
Users want to be able to favorite or unfavorite projects.

Blocked by #989 

There's a new action that comes up under the actions dropdown on the project detail page. If the user has the project favorited, the action will say unfavorite with a filled in star next to it. If the user doesn't have the project favorited, the action will say favorite with a not filled in start next to it. These actions will use the newly added favorite project endpoint on the backend (the frontend hookups, hooks, api calls, etc. will need to be added for the favorite project endpoint)

Add hooks, api calls, etc. to the frontend to call the favorite/unfavorite endpoints if they haven't been added by another ticket. Ask in #software if you have questions about what that means.

![image](https://user-images.githubusercontent.com/29521172/225202914-a6890766-a4ca-46b4-8c3a-4bdfd556e6a4.png)
![image](https://user-images.githubusercontent.com/29521172/225202936-ef063b5a-3a18-424f-b75f-1344922b5b2f.png)
